lms/djangoapps/completion_api/models.py

```diff
@@ -214,6 +213,23 @@ def vertical(self):
         """
         return self.get_completions_in_category('vertical')
 
+    @staticmethod
+    def _calculate_mean(course_key, exclude_users=None):
+        """
+        returns the course progress in average
+        """
+        course_avg = 0
+        course_metadata = CourseAggregatedMetaData.get_from_id(course_key)
+        total_possible_completions = float(course_metadata.total_assessments)
+        total_actual_completions = StudentProgress.get_total_completions(course_key, exclude_users=exclude_users)
+
+        total_users_qs = CourseEnrollment.objects.users_enrolled_in(course_key).exclude(id__in=exclude_users)
+        total_users = total_users_qs.count()
+        if total_users and total_actual_completions and total_possible_completions:
+            course_avg = total_actual_completions / float(total_users)
+            course_avg = min(100 * (course_avg / total_possible_completions), 100)  # avg in percentage
+        return course_avg
```

Comment:
Is this supposed to still be using the old progress metrics?  (Specifically, `StudentProgress` on L224, and `CourseMdouleCompletion` is still imported on L14). Might be fine.  Just clarifying how this fits in with other outstanding migration PRs.

Reply:
Oh nevermind.  I just saw that this is  in completion_api, which is itself transitional code.  Is this restoring code that was deleted from api_integration?

Reply:
Yes, this was importing code for calculating means from api-integration. That code is now adapted to work with the new completion API so it no longer works with the old API. So I've moved the original code here.